src/tools/miri/src/bin/miri.rs

```diff
@@ -59,15 +59,16 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
 
     fn after_analysis<'tcx>(
         &mut self,
+        handler: &EarlyErrorHandler,
```

Comment:
Instead of changing the signature of `after_analysis` for everyone, it would be better to just add the handler reference as a field in MiriCompilerCalls.

How does this even build currently? You haven't changed the call sites.

Reply:
This is a bit awkward if I try to add it as a field to `MiriCompilerCalls` since the inner handle of `handler` isn't `Sync`.